crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -142,15 +151,13 @@ impl Ethereum {
             .map_err(Error::Gas)
     }
 
-    fn into_request(tx: eth::Tx) -> web3::types::TransactionRequest {
-        web3::types::TransactionRequest {
-            from: tx.from.into(),
-            to: Some(tx.to.into()),
-            value: Some(tx.value.into()),
-            data: Some(tx.input.into()),
-            access_list: Some(tx.access_list.into()),
-            ..Default::default()
-        }
+    pub async fn balance(&self, address: eth::Address) -> Result<eth::Ether, Error> {
```

Comment:
nit: This could also be interpreted as the total balance of some random token address. Maybe mention eth in the function name or a doc comment.

Reply:
I'll add it as a doc-comment.

I feel like the name + type made it clear that it returned an Ether balance (specifically `balance() -> eth::Ether` instead of `balance() -> eth::Asset`).